### PR TITLE
Add IEEE 754 special-value tests for Abs

### PR DESCRIPTION
**Pull request by tphakala**
## Summary

- Add `TestAbs_SpecialValues` to both f32 and f64 packages
- Tests NaN, -NaN, +Inf, -Inf, -0, denormals, and max float values
- Verifies sign bit is cleared correctly for all IEEE 754 edge cases
- Follow-up from review feedback on PR #17

## Test plan

- [x] `go test ./f32/ ./f64/ -count=1` passes
- [x] `golangci-lint run ./f32/ ./f64/` — 0 issues

## Summary by CodeRabbit

* **Tests**
  * Enhanced test coverage for the floating-point absolute value function to thoroughly validate behavior on IEEE-754 special cases, including infinity, zero variants, subnormal values, and NaN handling.

### CODE REVIEW

File: f32/f32_test.go [hunk 1]
```diff
@@ -101,6 +101,60 @@ func TestAbs(t *testing.T) {
...
+		{"neg_max_float32", -math.MaxFloat32, math.MaxFloat32},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			dst := make([]float32, 1)
+			Abs(dst, []float32{tt.in})
+			if dst[0] != tt.want {
+				t.Errorf("Abs(%v) = %v, want %v", tt.in, dst[0], tt.want)
+			}
```
[the remark this answers is not in the document]
Comment (tphakala) [reply f32/f32_test.go]: Fixed in c9c5429 — using math.Float32bits/Float64bits for bit-level comparison.

File: f64/f64_test.go [hunk 2]
```diff
@@ -179,6 +179,60 @@ func TestAbs(t *testing.T) {
...
+		{"neg_max_float64", -math.MaxFloat64, math.MaxFloat64},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			dst := make([]float64, 1)
+			Abs(dst, []float64{tt.in})
+			if dst[0] != tt.want {
+				t.Errorf("Abs(%v) = %v, want %v", tt.in, dst[0], tt.want)
+			}
```
[the remark this answers is not in the document]
Comment (tphakala) [reply f64/f64_test.go]: Fixed in c9c5429 — using math.Float32bits/Float64bits for bit-level comparison.

File: f32/f32_test.go [hunk 3]
```diff
@@ -101,6 +101,60 @@ func TestAbs(t *testing.T) {
 	}
 }
 
+func TestAbs_SpecialValues(t *testing.T) {
+	posInf := float32(math.Inf(1))
+	negInf := float32(math.Inf(-1))
+	nan := float32(math.NaN())
+	negZero := float32(math.Float64frombits(1 << 63)) // -0
```
[the remark this answers is not in the document]
Comment (tphakala) [reply f32/f32_test.go]: Fixed in c9c5429 — using math.Float32frombits(1 << 31) directly.

File: f32/f32_test.go [hunk 4]
```diff
@@ -101,6 +101,60 @@ func TestAbs(t *testing.T) {
...
+		negNaN := math.Float32frombits(math.Float32bits(nan) | (1 << 31))
+		dst := make([]float32, 1)
+		Abs(dst, []float32{negNaN})
+		if !math.IsNaN(float64(dst[0])) {
+			t.Errorf("Abs(-NaN) = %v, want NaN", dst[0])
+		}
+		if math.Signbit(float64(dst[0])) {
+			t.Errorf("Abs(-NaN) has sign bit set, want positive NaN")
+		}
+	})
```
[the remark this answers is not in the document]
Comment (tphakala) [reply f32/f32_test.go:156]: With only two NaN cases, a separate table adds indirection without reducing much duplication. The current structure is clear enough.

File: f64/f64_test.go [hunk 5]
```diff
@@ -179,6 +179,60 @@ func TestAbs(t *testing.T) {
...
+		negNaN := math.Float64frombits(math.Float64bits(nan) | (1 << 63))
+		dst := make([]float64, 1)
+		Abs(dst, []float64{negNaN})
+		if !math.IsNaN(dst[0]) {
+			t.Errorf("Abs(-NaN) = %v, want NaN", dst[0])
+		}
+		if math.Signbit(dst[0]) {
+			t.Errorf("Abs(-NaN) has sign bit set, want positive NaN")
+		}
+	})
```
[the remark this answers is not in the document]
Comment (tphakala) [reply f64/f64_test.go:234]: With only two NaN cases, a separate table adds indirection without reducing much duplication. The current structure is clear enough.

File: f32/f32_test.go [hunk 6]
```diff
@@ -101,6 +101,61 @@ func TestAbs(t *testing.T) {
...
+}
```
[the remark this answers is not in the document]
Comment (tphakala) [reply f32/f32_test.go:157]: NaN tests can't use the same comparison as the table (bit equality doesn't work for NaN since NaN \!= NaN). The separate structure with math.IsNaN + math.Signbit checks is intentional.

File: f64/f64_test.go [hunk 7]
```diff
@@ -179,6 +179,61 @@ func TestAbs(t *testing.T) {
...
+}
```
[the remark this answers is not in the document]
Comment (tphakala) [reply f64/f64_test.go:235]: NaN tests can't use the same comparison as the table (bit equality doesn't work for NaN since NaN \!= NaN). The separate structure with math.IsNaN + math.Signbit checks is intentional.
